core/src/main/java/org/elasticsearch/bootstrap/JNACLibrary.java

```diff
@@ -39,6 +39,7 @@
     public static final int MCL_CURRENT = 1;
     public static final int ENOMEM = 12;
     public static final int RLIMIT_MEMLOCK = Constants.MAC_OS_X ? 6 : 8;
+    public static final int RLIMIT_NPROC = 6;
```

Comment:
can we please put a comment above this that it is a non-standard constant and confusingly on linux, the only place we are using it, it does not in fact mean number of processes, but number of threads.